astropy/stats/sigma_clipping.py

```diff
@@ -144,10 +144,19 @@ def _perform_clip(self, _filtered_data, axis=None):
         min_value = max_value - std * self.sigma_lower
         max_value += std * self.sigma_upper
 
+        # Ensure min/max can be broadcast with the data (if arrays):
         if axis is not None:
-            if axis != 0:
-                min_value = np.expand_dims(min_value, axis=axis)
-                max_value = np.expand_dims(max_value, axis=axis)
+            if not hasattr(axis, '__iter__'):
```

Comment:
I see from @saimn's post in #7227 that there's also an `isiterable()` function in `astropy.utils`.

Reply:
I've changed that in my copy (probably best not push every commit individually)...